app/controllers/admin/reports_controller.rb

```diff
@@ -66,5 +68,11 @@ def render_report_as(format)
         @report.render_as(format)
       end
     end
+
+    def render_timeout_error
+      assign_view_data
+      @error = ".report_taking_longer"
```

Comment:
I'm curious if there's any reason to translate in the view rather than here? I think we can access the translate helper in the controller.

This looks nice and clean, but I was thinking we usually expect translate keys to be with the `t()` method call, so if we ever need to rely on that pattern (eg when searching the codebase or perform mass find/replace), this could slow us down or cause us to miss it.

It's a really minor point but I was curious :)

Reply:
> I'm curious if there's any reason to translate in the view rather than here?

Yes, Rails supports translations in the view better.

* We can use the lazy loading with the prefixed dot.
* We can simply use `t` instead of `I18n.t`.
* Views are concerned with how things are displayed. An error code can be translated or a different error code could trigger a modal and that should be the concern of the view, not the controller, I think.

You got a good point about the disassociation of translation key and the `t` call as well. And I'm not sure what to suggest about that. If we want to translate in the view then we could do something like `t(".report_taking_longer") if @error == :report_taking_longer` but that seems cumbersome, too. So we can just translate in the controller and add an apt translation scope. My pattern is a bit closer to the translations of errors on models. There's a default structure of error messages on ActiveRecord errors. Maybe the path will become more clear later.

Reply:
Good points.